models/user.js

```diff
@@ -0,0 +1,49 @@
+var mongoose = require('mongoose')
+  , Schema = mongoose.Schema
+  , passportLocalMongoose = require('passport-local-mongoose');
+
+var UserSchema = new Schema({
+  username: {
+    type: String
+    , required: true
+    , lowercase: true // 'ABC' is equivalent 'abc'
+    , trim: true // ' abc ' will be trimmed to 'abc'
+    , index: { unique: true }
+  },
+  password: {
```

Comment:
For `User.register(new User({ username: req.body.username }), req.body.password, function(err)`, this username is fixed and will be save to db directly. 

This is how a User looks like in mongodb:

`{ "_id" : ObjectId("5569050f59fc9edea1372d00"), "salt" : "684fcbed7f80cde1005e7b27abab26f3f6d0aab7554c6d86b07d57908fb2ac85", "hash" : "a0a98023418ca9df5c062d411c9a37a03e47062ca96b188233c38a7da81245901868f6d0ba70c34e43a9f4aa0d89e36886aa7fbedb3b9040efcaca04f049f51977535cae222abf9e74a48e3f8b6dcdd1281cc598f9a7bd75863b84bdf655f23e4e2e78485f3514ab0383738d02dec881e169afe4fa41bb3e2019b23a9cb6564970421c0fc5b49ae91a96c7c5fdf702fd85ea75c2d9a9bb92d676a52f9a18e16b5ef5e39d70c869468e7de243460b8a1990d1ecff9fe14721004292f0d8520db2696bc713af3681ccf85b44491502836332c07d0b1d3546c09576eac18ea751aa7c0b157026f22b2d949dba176d164e82812b223718f55ef8a9cca35e801e5b45f064035e3b0c8175a459d34a5c2250cec24cc3aff96604d11065f78752eff433fbcb32fe2cf4e69a49de51b09500ea9f6b2e082685e8d03abebf5182ddb4dcde59173a16b072335dfe080a87a122187875c47fdd5b5f79aac333f0f7c0d2eccc49e23981aa843b0b3240d06fcf780ab773893e98deb5300875aeb390481388da70f7aa12076ba95105411fd7fbf54eeec42e4b10739c34941c76ab995041a2c91efff385bf3145768f65489f8dafbbdc54d72dffef7b011b4d5e7b737090681cc44abe2612742f5ae284e913f1501b49b9581397688414dc311bb99e93888bf703a2d6d63624d2795a60e6d9fcd6550e51105691506dfed448440db1ecb446b8", "username" : "admin", "locked" : false, "is_admin" : false, "banned" : false, "date_created" : ISODate("2015-05-30T00:32:15.905Z"), "__v" : 0 }`